docs: update docs for multi metrics provider support, fix API reference generator (#1251)
keptn/lifecycle-toolkit: docs/content/en/docs/concepts/metrics/_index.md

The metric will be collected from the provider specified in the

> specs.provider.name field.

Reviewer: ```suggestion
`specs.provider.name` field.
```
Or is this the sort of formatting that is ill-advised here?  If so, just reject my suggestion ;-)

The query is a string in the provider-specific query language, used to obtain a metric.

> Providing the metrics as CRDs in a K8s cluster will facilitate the reusability of this data across multiple components.

Reviewer: ```suggestion
Providing the metrics as CRDs in a K8s cluster facilitates the reusability of this data across multiple components.
```

> Furthermore, this allows using multiple observability platforms for different metrics at the same time.

Reviewer: Do you want to note that, at this time, only one occurence of each observability platform is allowed per namespace or is that overkill?

Author: No, with multi provider support that limitation is gone. You can now specify multiple of the same type :)

A `KeptnMetric` looks like the following:

```yaml
apiVersion: metrics.keptn.sh/v1alpha3
kind: KeptnMetric
metadata:
  name: keptnmetric-sample
  namespace: podtato-kubectl
spec:
  provider:
    name: "my-provider"
  query: "sum(kube_pod_container_resource_limits{resource='cpu'})"
  fetchIntervalSeconds: 5
```

In this example, the provider is set to `my-provider`.
The provider tells the metrics-operator where to get the value for the `KeptnMetric`, and its configuration looks follows:

```yaml
apiVersion: metrics.keptn.sh/v1alpha3
kind: KeptnMetricsProvider
metadata:
  name: my-provider
spec:
  type: prometheus
  targetServer: "http://prometheus-k8s.monitoring.svc.cluster.local:9090"
```

As you can see, the provider type is set to `prometheus`, which is one of the currently supported `KeptnMetricProviders`.
By using different names for different providers of the same type, you can fetch metrics from multiple
provider instances at the same time.

> Other supported provider types are `dynatrace`, and `dql`:

Reviewer 2: ```suggestion
Other supported provider types are `dynatrace`, `datadog`, and `dql`:
```

Author: good catch!
